server/startup/migrations/v099.js

```diff
@@ -0,0 +1,217 @@
+	console.log('[AVATAR]', ...args);
+}
+
+function logError(...args) {
+	console.error('[AVATAR]', ...args);
+}
+
+function insertAvatar({ details, avatarsFileStore, stream, callback = () => {} }) {
+	return new Promise((resolve) => {
+		Meteor.defer(() => {
+			Meteor.runAsUser('rocket.cat', () => {
+				avatarsFileStore.insert(details, stream, (err) => {
+					if (err) {
+						logError({err});
+						resolve();
+					} else {
+						Meteor.setTimeout(() => {
+							callback();
+						}, 200);
+					}
+					resolve();
+				});
+			});
+		});
+	});
+}
+
+function batch(arr, limit, fn) {
+	if (!arr.length) {
+		return Promise.resolve();
+	}
+	return Promise.all(arr.splice(0, limit).map((item) => {
+		return fn(item);
+	})).then(() => { return batch(arr, limit, fn); });
+}
+
+RocketChat.Migrations.add({
+	version: 99,
+	up() {
+		log('Migrating avatars. This might take a while.');
+
+		const query = {
+			$or: [{
+				's3.path': {
+					$exists: true
+				}
+			}, {
+				'googleCloudStorage.path': {
+					$exists: true
+				}
+			}]
+		};
+
+		RocketChat.models.Uploads.find(query).forEach((record) => {
+			if (record.s3) {
+				RocketChat.models.Uploads.model.direct.update({_id: record._id}, {
+					$set: {
+						'store': 'AmazonS3:Uploads',
+						AmazonS3: {
+							path: record.s3.path + record._id
+						}
+					},
+					$unset: {
+						s3: 1
+					}
+				}, {multi: true});
+			} else {
+				RocketChat.models.Uploads.model.direct.update({_id: record._id}, {
+					$set: {
+						store: 'GoogleCloudStorage:Uploads',
+						GoogleStorage: {
+							path: record.googleCloudStorage.path + record._id
+						}
+					},
+					$unset: {
+						googleCloudStorage: 1
+					}
+				}, {multi: true});
+			}
+		});
+
+		RocketChat.models.Uploads.model.direct.update({
+			store: 'fileSystem'
+		}, {
+			$set: {
+				store: 'FileSystem:Uploads'
+			}
+		}, {
+			multi: true
+		});
+		RocketChat.models.Uploads.model.direct.update({
+			store: 'rocketchat_uploads'
+		}, {
+			$set: {
+				store: 'GridFS:Uploads'
+			}
+		}, {
+			multi: true
+		});
+
+		const avatarOrigins = [
+			'upload',
+			'gravatar',
+			'facebook',
+			'twitter',
+			'github',
+			'google',
+			'url',
+			'gitlab',
+			'linkedin'
+		];
+
+		const avatarsPathRecord = RocketChat.models.Settings.findOne({_id: 'Accounts_AvatarStorePath'});
+		const avatarStoreTypeRecord = RocketChat.models.Settings.findOne({_id: 'Accounts_AvatarStoreType'});
+
+		const avatarsPath = avatarsPathRecord ? avatarsPathRecord.value : process.env.AVATARS_PATH;
+		let avatarStoreType = avatarStoreTypeRecord && avatarStoreTypeRecord.value;
+
+		const oldAvatarGridFS = new RocketChatFile.GridFS({
+			name: 'avatars'
+		});
+
+		// 1. Novo usuário, não executa migration
+		// 2. Antigo, migra de antes da 97, vai ter storage type
+		// 3. Executou a 97, não vai ter storeType
+		// 3.1 Se ainda tem dados no gridfs, manda ver
+		// 3.2 Se não tem, quebra e pede a variável OU forçar a reexecutar a 97???
```

Comment:
According to Google Translate.  Just for my own reference :grin: 
// 1. New user, does not execute migration
// 2. Old, migrated before 97, will have storage type
// 3. Has run to 97, will not have storeType
// 3.1 If you still have data in gridfs, you have to see
// 3.2 If it does not have, break and asks for the variable OR force to re-run to 97 ???